Feat: Bulk Upload in Asset Manager

cqliu1 opened the pull request:
Related to #23064.
Closes #28277.

This adds a file picker to the asset manager that allows you to upload multiple images to a workpad without going through an image element. With this, you can load up your workpad with all of the assets you want to use, and easily select them from the asset picker when you add your image elements.

![screen shot 2019-01-18 at 10 48 58 am](https://user-images.githubusercontent.com/1697105/51404016-6d140a00-1b0f-11e9-99b5-3452b5af6343.png)

![jan-18-2019 13-11-17](https://user-images.githubusercontent.com/1697105/51410682-9e4a0580-1b22-11e9-8996-5bb1c601081c.gif)

I also added an `accept` attribute to all the file pickers. Now file pickers in image upload and in the asset manager will only accept images, and the one in the workpad loader will only accept JSON files.

#### Workpad Loader
Only JSON files are selectable
![screen shot 2019-01-18 at 11 06 49 am](https://user-images.githubusercontent.com/1697105/51404618-363ef380-1b11-11e9-9f84-34a3c61788e2.png)

#### Image Upload
Only image files are selectable
![screen shot 2019-01-18 at 11 06 10 am](https://user-images.githubusercontent.com/1697105/51404589-15769e00-1b11-11e9-89be-31a04a523bee.png)

elasticmachine commented:
Pinging @elastic/kibana-canvas

elasticmachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pull-request/7999/)

w33ble commented:
We probably shouldn't hide the asset interface if you don't have anything uploaded anymore.

cqliu1 commented:
retest

w33ble commented:
What's up with the close button once you upload things? Can we disable that, or do we need to open an issue in EUI? 

![jan-18-2019 11-56-06](https://user-images.githubusercontent.com/404731/51407077-270f7400-1b18-11e9-9a40-28861c3ec4e0.gif)

w33ble commented:
Looks like this uploader doesn't de-dupe assets. When I upload the same image from the normal file uploader, it won't create a second asset, it'll reuse the one that's already uploaded:

![jan-18-2019 12-02-04](https://user-images.githubusercontent.com/404731/51407626-a81b3b00-1b19-11e9-8956-5d44c7b66630.gif)

But if I upload the same asset from the assets modal, it'll create a duplicate:

![jan-18-2019 12-03-02](https://user-images.githubusercontent.com/404731/51407647-b79a8400-1b19-11e9-8598-ff8195af5d74.gif)

elasticmachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pull-request/8006/)

cqliu1 commented:
> We probably shouldn't hide the asset interface if you don't have anything uploaded anymore.
Good call! I removed the conditional for rendering the asset manager button, and now it'll show up new workpads. 

I de-duped the upload from the asset manager. 

I also added a loading indicator to disable the filepicker when images are being saved.
![jan-18-2019 13-21-19](https://user-images.githubusercontent.com/1697105/51411127-fa615980-1b23-11e9-8535-dd53f2e000ae.gif)

elasticmachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pull-request/8025/)

cqliu1 commented:
I added an `EuiEmptyPrompt` for when you have no assets uploaded. 

![screen shot 2019-01-18 at 2 22 14 pm](https://user-images.githubusercontent.com/1697105/51413515-7a8bbd00-1b2c-11e9-8834-457fb333557a.png)

elasticmachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pull-request/8051/)

cqliu1 commented:
@ryankeairns I added the same style you applied to the file picker in the workpad loader, which makes the height taller, but I can undo that change. 

How does this look?
<img width="980" alt="screen shot 2019-01-25 at 5 00 03 pm" src="https://user-images.githubusercontent.com/1697105/51779165-f81e7280-20c2-11e9-854a-2526734ce9c6.png">

elasticmachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pull-request/9165/)

elasticmachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pull-request/9822/)
